docs/en/developers/04_Pull_requests.md

```diff
@@ -0,0 +1,71 @@
+$ git checkout master      # go on master branch
+$ git pull upstream master # pull the last version of master
+$ git checkout -           # go back to your branch
+$ git rebase master        # rebase your branch on master
+```
+
+If you feel confident, you can use `git rebase -i master` to rewrite your history and make it clearer.
+
+### Merging
+
+If you prefer, you can simply merge `master` into your own branch. Conflicts might be easier to resolve, but your Git history will be less readable. Don't worry, we'll take care of it before merging your PR back into `master`.
+
+To merge `master`:
+
+```console
+$ git checkout master      # go on master branch
+$ git pull upstream master # pull the last version of master
+$ git checkout -           # go back to your branch
+$ git merge master         # merge master into your branch
+```
+
+## How to write a Git commit message
+
+It is important to have proper commits messages in order to facilitate later debugging, and so to read the following advice. Commits messages should inform about the choices made in the past (the “why?”)
+
+The first line should start by a verb (e.g. "Add") and explain in few words the objective of the commit. It's usually less than 50 characters so it stays concise. You can consider this line as the subject of your commit.
+
+Then, let an empty line, and start to write the body. It's usually less than 72 characters, but you are pretty free in the tone of the message. The body is the place where you can give the context of your patch. For instance, you can explain what you were doing when you identified a bug, or the problem you had before your patch. Giving these information help other developers to understand why a specific choice was made, especially when a patch introduces a bug that is identified months later.
+
+You also can add a bunch of references (e.g. the URL to the initial ticket in the bug tracker, a reference to some forum explaining a point).
+
+You can find more information about commit messages [on this blog post](https://chris.beams.io/posts/git-commit/).
+
+## How to write tests
+
+FreshRSS has almost no tests for now, but we should improve things. We added this point to the checklist to help us to write more tests, and we would really appreciate if you help us by writing a test proving your patch is working.
+
+We're using [PHPUnit](https://phpunit.de/) version 7.5 ([documentation](https://phpunit.readthedocs.io/en/7.5/)).
+
+You'll find more information on how to run tests [in this document](03_Running_tests.md).
+
+Feel free to ask us for some guidance. Everything will not be easy to test, so don't spend too much time on this.
+
+## Why you should write documentation
+
+FreshRSS had for a long time a poor documentation. A friendly project must have a complete documentation though, so new comers don't have to ask too many questions, and users can find answers to their problems. The documentation should not be written “later” or it will never be.
```

Comment:
```suggestion
A friendly project should have correct and complete documentation, so newcomers don't have to ask too many questions, and users can find answers to their problems. The documentation should not be written “later” or chances are it'll never be.
```

No need to tear the project down publicly. ;-)

Reply:
You're right, I tend to be too dramatic!